client/packages/react/src/InstantReactAbstractDatabase.ts

```diff
@@ -57,6 +57,195 @@ export type TypingIndicatorHandle<PresenceShape> = {
 
 export const defaultActivityStopTimeout = 1_000;
 
+// ------
+// Topics
+
+function useTopicEffect<
+  RoomSchema extends RoomSchemaShape,
+  RoomType extends keyof RoomSchema,
+  TopicType extends keyof RoomSchema[RoomType]["topics"],
+>(
+  room: InstantReactRoom<any, RoomSchema, RoomType>,
+  topic: TopicType,
+  onEvent: (
+    event: RoomSchema[RoomType]["topics"][TopicType],
+    peer: RoomSchema[RoomType]["presence"],
+  ) => any,
+): void {
+  useEffect(() => {
+    const unsub = room._core._reactor.subscribeTopic(
+      room.id,
+      topic,
+      (event, peer) => {
+        onEvent(event, peer);
+      },
+    );
+
+    return unsub;
+  }, [room.id, topic]);
+}
+
+function usePublishTopic<
+  RoomSchema extends RoomSchemaShape,
+  RoomType extends keyof RoomSchema,
+  TopicType extends keyof RoomSchema[RoomType]["topics"],
+>(
+  room: InstantReactRoom<any, RoomSchema, RoomType>,
+  topic: TopicType,
+): (data: RoomSchema[RoomType]["topics"][TopicType]) => void {
+  useEffect(() => room._core._reactor.joinRoom(room.id), [room.id]);
```

Comment:
Another interesting thing here -- I didn't realize we _always_ sent `join-room` events, whenever we mounted a hook like this. I am not sure it's a problem, but stuff like this could explain why we see so many join-room events